ambassador/ambassador/ir/irauth.py

```diff
@@ -141,22 +127,14 @@ def _load_auth(self, module: Resource):
                     self[key] = value
 
             self.referenced_by(module)
+        
+        self.to_header_list("allowed_request_headers", module)
```

Comment:
We should still allow the v0 `AuthService` to work, which probably means... what, if there's no `allowed_request_headers`, we fall back to `allowed_headers` and merge the defaults in?

Reply:
Sure, by having `allowed_headers` in both `allowed_request_headers` and `allowed_authorization_headers`. I will make sure that it is well documented so whoever decides to still use `V0` and `allowed_headers` knows that this might produce undesirable behaviour.

Reply:
Right, that sounds like a great answer. I like `IRAuth` never having an `allowed_headers` attribute and always having an `allowed_request_headers` attribute.

Reply:
Just to be clear here. I'm not suggesting to have `allowed_headers` in both `allowed_request_headers` and `allowed_authorization_headers`, that's how it was implemented before this PR. I just assumed that the logic should remain the same.